test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,756 @@
+				gatewayLbService  = "gateway-openshift-default"
+				gatewayName       = "gateway"
+			)
+
+			// ensure default gateway objects is created
+			coreClient := clientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Getting the default domain")
+			defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+			defaultDomain := strings.Split(defaultIngressDomain, "apps.")[1]
+
+			g.By("Create the default API Gateway")
+			createGateway(oc, gatewayName, gatewayClassName, defaultDomain)
+
+			g.By("Ensure the gateway's LoadBalancer service and DNSRecords are available")
+			gwlbIP, err := ensureLbServiceRetrieveLbIPOrFqdn(oc, ingressNamespace, gatewayLbService)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			gwwAddress, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", "openshift-ingress", "gateway", gatewayName, "-o=jsonpath={.status.addresses[0].value}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.Logf("The gateway Aaddress is: %v", gwwAddress)
+			o.Expect(gwlbIP).To(o.Equal(gwwAddress))
+
+			// get the dnsrecord name
+			dnsRecordName, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", "openshift-ingress", "dnsrecord", "-l", "gateway.networking.k8s.io/gateway-name="+gatewayName, "-o=jsonpath={.items[*].metadata.name}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.Logf("The gateway API dnsrecord name is: %v", dnsRecordName)
+			// check whether status of dns reccord is True
+			dnsRecordstatus, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", "openshift-ingress", "dnsrecord", dnsRecordName, `-o=jsonpath={.status.zones[0].conditions[0].status}`).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(dnsRecordstatus).To(o.Equal("True"))
+
+			// deleted the gateway loadbalancer service and then checked if it was restored
+			g.By(fmt.Sprintf("Try to delete the gateway lb service %s", gatewayLbService))
+			lbService, err := coreClient.CoreV1().Services(ingressNamespace).Get(context.Background(), gatewayLbService, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			createdTime1 := lbService.ObjectMeta.CreationTimestamp
+			err = oc.AdminKubeClient().CoreV1().Services(ingressNamespace).Delete(context.Background(), gatewayLbService, metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By(fmt.Sprintf("Wait until the gateway lb service %s is automatically recreated successfully", gatewayLbService))
+			o.Eventually(func() bool {
+				lbService, err := coreClient.CoreV1().Services(ingressNamespace).Get(context.Background(), gatewayLbService, metav1.GetOptions{})
+				if err != nil {
+					if errors.IsNotFound(err) {
+						return false
+					}
+					e2e.Logf("Error getting the gateway lb service %s: %v, retrying...", gatewayLbService, err)
+					return false
+				}
+
+				createdTime2 := lbService.ObjectMeta.CreationTimestamp
```

Comment:
Do we need to check createdTime ? Are you sure you could see the condition `createdTime2 == createdTime1` ?

Reply:
Yes, for aws, the lb service has external hostname(not ip). the update is very quick. Once the test failed, and I added checking the created time.

```
    STEP: try to delete the gateway lb service gateway-openshift-default @ 04/09/25 11:20:01.711
    STEP: wait until the gateway lb service gateway-openshift-default is automatically recreated successfully @ 04/09/25 11:20:02.139
  I0409 11:20:15.361644 59042 gatewayapicontroller.go:350] Failed to get the new IP or hostname of the gateway lb service gateway-openshift-default, try next round
  I0409 11:20:18.575370 59042 gatewayapicontroller.go:346] new load balancer hostname a762f66609bac406eb7c213232567be5-251282266.us-east-2.elb.amazonaws.com is available
```